synapse/util/caches/response_cache.py

```diff
@@ -63,10 +74,7 @@ class ResponseCache(Generic[KV]):
     """
 
     def __init__(self, clock: Clock, name: str, timeout_ms: float = 0):
-        # This is poorly-named: it includes both complete and incomplete results.
-        # We keep complete results rather than switching to absolute values because
-        # that makes it easier to cache Failure results.
-        self.pending_result_cache: Dict[KV, ObservableDeferred] = {}
+        self.result_cache: Dict[KV, ResponseCacheEntry] = {}
```

Comment:
since we're changing the type, we may as well take the opportunity to fix the name.